### PR TITLE
MIGENG-328 Added 'ReasonableDefaults' rules

### PR DESCRIPTION
**Pull request by mrizzi**
https://issues.redhat.com/browse/MIGENG-328

Changes:

- new set of three rules to provide `ReasonableDefaults` for `VMWorkloadInventoryModel` optional input fields `datacenter`, `cluster` and `host_name` with `ReasonableDefaultsTest` tests
- changed accordingly both`BasicFieldsTest` (adding `testWithoutOptionalInputFields` test) and `WorkloadInventoryReportTest`
- the default values are in the `WorkloadInventoryReportModel` class
- introduced `createAndExecuteCommandsAndGetResults` and `extractModels` methods to be used in all tests that extends `BaseTest` (and refactored tests)

### CODE REVIEW

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java [hunk 1]
```diff
@@ -25,9 +25,14 @@ public BasicFieldsTest() {
         super("/org/jboss/xavier/analytics/rules/workload/inventory/BasicFields.drl", ResourceType.DRL);
     }
 
-    @Test
-    public void test() throws ParseException {
+    private void checkLoadedRulesNumber()
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java]: What do you think about annotating with @Before as you are running it before each test.

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java [hunk 2]
```diff
@@ -25,9 +25,14 @@ public BasicFieldsTest() {
...
+    {
         Utils.checkLoadedRulesNumber(kieSession, "org.jboss.xavier.analytics.rules.workload.inventory", 1);
+    }
+
+    @Test
+    public void testWithAllInputFields() throws ParseException {
+        checkLoadedRulesNumber();
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java]: if you add the @before method then you can remove this line

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java [hunk 3]
```diff
@@ -80,4 +76,48 @@ public void test() throws ParseException {
         Assert.assertEquals("esx13.v2v.bos.redhat.com", report.getHost_name());
         Assert.assertEquals(new SimpleDateFormat("yyyy-M-dd'T'hh:mm:ss.S").parse("2019-09-18T14:52:45.871Z"), report.getCreationDate());
     }
+
+    @Test
+    public void testWithoutOptionalInputFields() throws ParseException {
+        checkLoadedRulesNumber();
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java]: if you add the @before method then you can remove this line

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/ReasonableDefaultsTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    private void checkLoadedRulesNumber()
+    {
+        Utils.checkLoadedRulesNumber(kieSession, "org.jboss.xavier.analytics.rules.workload.inventory", 3);
+    }
+
+    @Test
+    public void testDatacenterFieldNullValueShouldFireRule()
+    {
+        checkLoadedRulesNumber();
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/ReasonableDefaultsTest.java]: you can extract this check with a @Before method

File: src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java [hunk 5]
```diff
@@ -43,4 +47,13 @@ public void setup()
     public void tearDown()
     {
     }
+
+    public <T> List<T> extractModels(Map<String, Object> results, Class<T> model)
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java]: This code is duplicated in BaseTest. It could be refactored and extract a common super class in order to reduce duplication ?

File: src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java [hunk 6]
```diff
@@ -43,4 +47,13 @@ public void setup()
     public void tearDown()
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java]: I see it's not part of this PR, but why this empty method ?

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java (hunk 1 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java]: Thanks for the idea: it made me think about this "check" and i decided to "promoted" it to become a test

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java (hunk 3 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java]: See above.

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/ReasonableDefaultsTest.java (hunk 4 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/ReasonableDefaultsTest.java]: See above.

File: src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java (hunk 5 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java]: This rang a bell in my head since i was sure about having already implemented this method and, in fact, it was in the `Utils` class so i refactored for having that to be used

File: src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java (hunk 6 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/BaseIntegrationTest.java]: Removed :+1:

File: src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java (hunk 2 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/analytics/rules/workload/inventory/BasicFieldsTest.java]: See above.

**Comment by mrizzi** [comment]
@jonathanvila changes applied.
I still believe some refactoring between `BaseTest` and `BaseIntegrationTest` can be done but i don't want to get this PR to go too much off topic so i will raise a Jira for improving the refactoring.
